src/input_processing/dual_axis/bounds.rs

```diff
@@ -0,0 +1,299 @@
+//! Value bounds for dual-axis inputs
+
+use std::hash::{Hash, Hasher};
+
+use bevy::prelude::*;
+use bevy::utils::FloatOrd;
+use serde::{Deserialize, Serialize};
+
+use super::super::single_axis::*;
+use super::DualAxisProcessor;
+use crate::define_dual_axis_processor;
+
+define_dual_axis_processor!(
```

Comment:
No macros please. I don't think the need is compelling enough to justify the serious cost to maintainability.